numba/np/arrayobj.py

```diff
@@ -1131,6 +1150,12 @@ def fancy_getitem(context, builder, sig, args,
 
     indexer.end_loops()
 
+    for _indexer in indexer.indexers:
+        if isinstance(_indexer, IntegerArrayIndexer) \
+           and hasattr(_indexer, "idxary_instr"):
+            context.nrt.decref(builder, _indexer.idxty,
+                               _indexer.idxary_instr)
```

Comment:
As per a later comment, this _requires_ the presence of the NRT which isn't guaranteed and needs guarding against.

Reply:
Resolving this can be postponed as the 3 points in https://github.com/numba/numba/pull/8491#pullrequestreview-1411169708 will cover it.